pkg/project/registry/projectrequest/delegated/delegated.go

```diff
@@ -133,18 +133,18 @@ func (r *REST) Create(ctx apirequest.Context, obj runtime.Object, includeUniniti
 
 	// one of the items in this list should be the project.  We are going to locate it, remove it from the list, create it separately
 	var projectFromTemplate *projectapi.Project
-	var lastRoleBinding *authorizationapi.RoleBinding
+	lastRoleBindingName := ""
 	objectsToCreate := &kapi.List{}
 	for i := range list.Objects {
```

Comment:
```go
for _, lobj := range list.Objects {
	switch t := lobj.(type) {
	case *projectapi.Project:
		projectFromTemplate = t
		// don't add this to the list to create.  We'll create the project separately.
		continue
	case *rbac.RoleBinding:
		lastRoleBindingName = t.Name
	case *authorizationapi.RoleBinding:
		lastRoleBindingName = t.Name
	default:
		// no-op we only skip creation of project
	}

	objectsToCreate.Items = append(objectsToCreate.Items, lobj)
}
```

@liggitt any reason we were doing `list.Objects[i]` before?  Also, do we want to error if we see more than one project in the template?

Reply:
>@liggitt any reason we were doing list.Objects[i] before? 

so we didn't have to reason about range variable reuse outside the loop

>  Also, do we want to error if we see more than one project in the template?

probably

Reply:
> so we didn't have to reason about range variable reuse outside the loop

At worst it would only shadow a var outside the loop (I picked a unique name to avoid that as well)?

Reply:
at worst, you'd reuse a memory address and actually append the same item to the list of items to create N times

Reply:
> at worst, you'd reuse a memory address and actually append the same item to the list of items to create N times

Ah you are correct, I always forget the "don't use range vars + append together".

```go
for i := range list.Objects {
	switch t := list.Objects[i].(type) {
	case *projectapi.Project:
		projectFromTemplate = t
		// don't add this to the list to create.  We'll create the project separately.
		continue
	case *rbac.RoleBinding:
		lastRoleBindingName = t.Name
	case *authorizationapi.RoleBinding:
		lastRoleBindingName = t.Name
	default:
		// no-op we only skip creation of project
	}
	// use list.Objects[i] in append to avoid range memory address reuse
	objectsToCreate.Items = append(objectsToCreate.Items, list.Objects[i])
}
```